presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -577,15 +581,21 @@ public boolean transitionToFinishing()
         }
 
         if (autoCommit) {
-            transactionManager.asyncCommit(session.getTransactionId().get())
-                    .whenComplete((value, throwable) -> {
-                        if (throwable == null) {
-                            transitionToFinished();
-                        }
-                        else {
-                            transitionToFailed(unwrapCompletionException(throwable));
-                        }
-                    });
+            ListenableFuture<?> commitFuture = transactionManager.asyncCommit(session.getTransactionId().get());
+            Futures.addCallback(commitFuture, new FutureCallback<Object>()
+            {
+                @Override
+                public void onSuccess(@Nullable Object result)
+                {
+                    transitionToFinished();
+                }
+
+                @Override
+                public void onFailure(Throwable throwable)
+                {
+                    transitionToFailed(unwrapCompletionException(throwable));
+                }
+            });
```

Comment:
There is only one thing I miss about CompletableFuture and that is the fluent interface.  I currently don't missis it enough to spend the couple of hours it would take to write a `FluentFuture` transformer for ListenableFuture, but if enough people complained about it I would.  

As for the reason to switch from CompletableFuture, the problem is it is crazy difficult to use properly.  When you are using it you are not programming for a future value, but programming a completion stage.  This is no easy to check for completeness, and it doesn't support cancelation.  Also, it is always a mutable value.  As i mentioned earlier, when anyone starts a project that touches the async CompletableFuture code there is a long confusing discussion about the exact semantics for the API.  With guava settable future, we don't have any of these issues, and the cost is lack of a fluent API can can easily be addressed.